sycl-jit/jit-compiler/lib/KernelFusion.cpp

```diff
@@ -35,7 +35,16 @@ static JITResult errorToFusionResult(llvm::Error &&Err,
                           // compiled without exception support.
                           ErrMsg << "\t" << StrErr.getMessage() << "\n";
                         });
-  return JITResult{ErrMsg.str().c_str()};
+  return ResultType{ErrMsg.str().c_str()};
+}
+
+static JITResult errorToFusionResult(llvm::Error &&Err,
+                                     const std::string &Msg) {
+  return wrapError<JITResult>(std::move(Err), Msg);
+}
+
+static RTCResult errorToRTCResult(llvm::Error &&Err, const std::string &Msg) {
+  return wrapError<RTCResult>(std::move(Err), Msg);
 }
```

Comment:
Why don't we use `wrapError` directly (maybe with a different name)?

So, instead of `errorToFusionResult` or `errorToRTCResult`, we would call `errorToResult<JITResult>` and `errorToResult<RTCResult>`?

Reply:
`errorToResult<T>` makes sense, I initially decided against that to keep the diff a bit cleaner.